### ISSUE TITLE
Futures build fails with Boost 1.61 on non-Mac non-Mobile

### ISSUE THREAD

**Issue by tedjp** [facebook/folly#463]
Boost 1.61 modified boost::context and moved some things into boost::context::detail. Trying to build folly with Boost 1.61 therefore fails on non-Mac non-Mobile builds because Futures tries to use folly Fibers which are unavailable.

```
libtool: compile:  g++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/include -std=gnu++1y -g -O2 -MT futures/Barrier.lo -MD -MP -MF futures/.deps/Barrier.Tpo -c futures/Barrier.cpp  -fPIC -DPIC -o futures/.libs/Barrier.o
In file included from ./../folly/fibers/Fiber.h:27:0,
                 from ./../folly/fibers/Baton-inl.h:16,
                 from ./../folly/fibers/Baton.h:191,
                 from ./../folly/futures/Future-inl.h:35,
                 from ./../folly/futures/Future.h:506,
                 from ./../folly/futures/Barrier.h:22,
                 from futures/Barrier.cpp:17:
./../folly/fibers/BoostContextCompatibility.h:18:38: fatal error: boost/context/fcontext.hpp: No such file or directory
 #include <boost/context/fcontext.hpp>
                                      ^
compilation terminated.
```

Build continues by preventing the use of Fibers if boost::context is unavailable:
```
diff --git a/folly/futures/Future-inl.h b/folly/futures/Future-inl.h
index 0d17100..03b3657 100644
--- a/folly/futures/Future-inl.h
+++ b/folly/futures/Future-inl.h
@@ -28,7 +28,7 @@
 #include <folly/futures/detail/Core.h>
 #include <folly/futures/Timekeeper.h>
 
-#if FOLLY_MOBILE || defined(__APPLE__)
+#if !HAVE_BOOST_CONTEXT || (FOLLY_MOBILE || defined(__APPLE__))
 #define FOLLY_FUTURE_USING_FIBER 0
 #else
 #define FOLLY_FUTURE_USING_FIBER 1
```

**Comment by tedjp** [comment]
This is also a fix when compiling with older versions of Boost (1.55?) that also don't have boost::context.

**Comment by zonr** [comment]
Seems a duplicate of #413.

**Comment by wrigleyster** [comment]
Indeed it does, but it is not. The fix for #413, anyhow, does not fix this issue.
